client/dashboard/components/page-header/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { Breadcrumbs } from '@automattic/components/src/breadcrumbs';
+import {
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+} from '@wordpress/components';
+import type { PageHeaderProps } from './types';
+
+import './style.scss';
+
+/**
+ * The PageHeader component provides a structured introduction to a page or section,
+ * combining a title, optional description, and contextual actions. It supports
+ * varying levels of hierarchy through semantic heading levels, and can include
+ * visual decorations, navigational aids like breadcrumbs, and utility controls
+ * such as buttons or dropdowns.
+ *
+ * ```jsx
+ * import { PageHeader } from '@automattic/components';
+ * import { Button } from '@wordpress/components';
+ * import { cog } from '@wordpress/icons';
+ *
+ * function MyComponent() {
+ * 	return (
+ * 		<PageHeader
+ * 			title="Settings"
+ * 			description="Configure your application settings"
+ * 			decoration={<Icon icon={cog} />}
+ * 			actions={<Button variant="primary">Save Changes</Button>}
+ * 		/>
+ * 	);
+ * }
+ * ```
+ */
+export const PageHeader = ( {
+	title,
+	description,
+	actions,
+	decoration,
+	breadcrumbItems,
```

Comment:
Should this be just breadcrumb and actually be the whole component (composition) instead of passing the breadcrumbItems. Imagine we'd want more Breadcrumb props later.

Reply:
I don't think we need that much flexibility for a consumer to pass their own breadcrumb component, do we?

> Imagine we'd want more Breadcrumb props later.

`breadcrumbItems` are the exact type of the breadcrumb component items, so consumers would still need to update the passed props. Unless, I misunderstood your comment..

Reply:
This is not consistent with how "actions" work for instance. If "actions" accept "components" (aka `Buttons`) instead of buttonProps, why `breadcrumb` doesn't accept `Breadcrumb` and accept `breadcrumbProps` instead?

Reply:
Personally I don't think breadcrumbs and actions share that many similarities to be compared. It's true that `actions` here use composition and the design consistency can easily diverge from page to page, but this is the tradeoff for handling many different cases (buttons, menus, inputs, etc..) that will be needed in practice. 

Breadcrumbs on the other hand, I think we can be opinionated about the design and use the same Breadcrumbs(internally as is now) component everywhere. We expect a single component there always.

Reply:
If `compact` is applied, `mobileItem` can be provided to adapt the breadcrumb in mobile. How do I do that if the need arise and we don't want the "back" button?

You can argue that the need doesn't exist today but it's really not out of the question, and I'm 100% certain if the need shows up and we shipped `breadcrumbItems`, the developer would be templated to add `breadcrumbMobileItem` prop to `PageHeader` while the right and simple solution is just to do composition.

Reply:
`mobileItem` is an optional prop to Breadcrumb (I was just checking the README of that component) and if not provided it just renders and back button.

Reply:
I may have been looking on the wrong "breadcrumb" component (why we have two 😬 ). But the point is similar even though the new component doesn't seem to support any additional prop at the moment.

Reply:
Trying to follow the discussion... if we're not sure then why not just separate `<Breadcrumbs />` and `<PageHeader />` altogether. If a page needs both then we can just put them one after the other in `<PageLayout />`, similar to my original PR here: 

https://github.com/Automattic/wp-calypso/pull/103294/files#diff-baed26095b2d76890ebf83aa2a72893a9bbb1386c9584c88b60e706651592913R35

Reply:
^ Ignore if it's already an agreed API.

Reply:
>  if we're not sure then why not just separate <Breadcrumbs /> and <PageHeader /> altogether. If a page needs both then we can just put them one after the other in <PageLayout />, similar to my original PR here:

One counter argument is that "breadcrumb" could be consider part of the "header" and needs to be spaced properly with the title... But both options work for me for now.